rtengine/gauss.cc

```diff
@@ -0,0 +1,1293 @@
+                    gauss3x3mult<T> (src, dst, W, H, c0, c1, c2, b0, b1);
+                    break;
+
+                case GAUSS_DIV      :
+                    gauss3x3div<T> (src, dst, buffer2, W, H, c0, c1, c2, b0, b1);
+                    break;
+
+                case GAUSS_STANDARD :
+                    gauss3x3<T> (src, dst, W, H, c0, c1, c2, b0, b1);
+                    break;
+                }
+            } else {
+                // compute kernel values for separated 3x3 gaussian blur
+                double c1 = exp (-1.0 / (2.0 * sigma * sigma));
+                double csum = 2.0 * c1 + 1.0;
+                c1 /= csum;
+                double c0 = 1.0 / csum;
+                gaussHorizontal3<T> (src, dst, W, H, c0, c1);
+                gaussVertical3<T>   (dst, dst, W, H, c0, c1);
+            }
+        } else {
+#ifdef __SSE2__
+
+            if (sigma < GAUSS_DOUBLE) {
+                switch (gausstype) {
+                case GAUSS_MULT : {
+                    gaussHorizontalSse<T> (src, src, W, H, sigma);
+                    gaussVerticalSsemult<T> (src, dst, W, H, sigma);
+                    break;
+                }
+
+                case GAUSS_DIV : {
+                    gaussHorizontalSse<T> (src, dst, W, H, sigma);
+                    gaussVerticalSsediv<T> (dst, dst, buffer2, W, H, sigma);
+                    break;
+                }
+
+                case GAUSS_STANDARD : {
+                    gaussHorizontalSse<T> (src, dst, W, H, sigma);
+                    gaussVerticalSse<T> (dst, dst, W, H, sigma);
+                    break;
+                }
+                }
+            } else { // large sigma only with double precision
+                gaussHorizontal<T> (src, dst, W, H, sigma);
+                gaussVertical<T>   (dst, dst, W, H, sigma);
+            }
+
+#else
+            if (sigma < GAUSS_DOUBLE) {
+                switch (gausstype) {
+                case GAUSS_MULT : {
+                    gaussHorizontal<T> (src, src, W, H, sigma);
+                    gaussVerticalmult<T> (src, dst, W, H, sigma);
+                    break;
+                }
+
+                case GAUSS_DIV : {
+                    gaussHorizontal<T> (src, dst, W, H, sigma);
+                    gaussVerticaldiv<T> (dst, dst, buffer2, W, H, sigma);
+                    break;
+                }
+
+                case GAUSS_STANDARD : {
+                    gaussHorizontal<T> (src, dst, W, H, sigma);
+                    gaussVertical<T> (dst, dst, W, H, sigma);
+                    break;
+                }
+                }
+            } else { // large sigma only with double precision
+                gaussHorizontal<T> (src, dst, W, H, sigma);
+                gaussVertical<T>   (dst, dst, W, H, sigma);
+            }
+#endif
+        }
+    }
+}
+
+void gaussianBlur(float** src, float** dst, const int W, const int H, const double sigma, float *buffer, eGaussType gausstype, float** buffer2)
```

Comment:
Since `gaussianBlur` is the only function that should be visible in other translation units, everything else in the source file should probably go into the anonymous namespace to tell the compiler and linker that all generated code should have static visibility which will improve optimization possibilities.